ui/summary.go

```diff
@@ -384,3 +385,44 @@ func (s *Summary) SummarizeMetrics(w io.Writer, indent string, data SummaryData)
 
 	s.summarizeMetrics(w, indent+"  ", data.Time, data.TimeUnit, data.Metrics)
 }
+
+func newJSONEncoder(w io.Writer) *json.Encoder {
+	encoder := json.NewEncoder(w)
+	encoder.SetIndent("", "    ")
+	return encoder
+}
+
+func summarizeGroupJSON(w io.Writer, group *lib.Group) error {
+	return newJSONEncoder(w).Encode(group)
+}
+
+func summarizeMetricsJSON(w io.Writer, t time.Duration, timeUnit string, metrics map[string]*stats.Metric) error {
+	data := make(map[string]interface{})
+	for name, m := range metrics {
+		m.Sink.Calc()
+		if sink, ok := m.Sink.(*stats.TrendSink); ok {
+			data[name] = sink.Format(t)
+			continue
+		}
+		data[name] = m.Sink.Format(t)
+		_, extra := nonTrendMetricValueForSum(t, timeUnit, m)
+		if len(extra) > 1 {
+			extraData := make(map[string]interface{})
+			extraData["value"] = m.Sink.Format(t)["value"]
```

Comment:
Is there a reason for calling `m.Sink.Format(t)` again here? Would this be equivalent to `extraData["value"] = data[name]["value"]`? Or is this intentional and we want these to be separate instances because of the JSON encoder?

Reply:
If it's the extra field, we overwrite the original `data[name]`.

Reply:
I don't see how that's related, as that happens a couple of lines below this.

My point is related to the other comment of `Format(t)` being called 3 times. I'm wondering if it's possible to call it once and use the same result for all.

Reply:
Fixed